src/modules/FunModule.cpp

```diff
@@ -0,0 +1,61 @@
+#include "FunModule.h"
+#include <QtNetwork>
+#include <QtCore/QFile>
+#include <QtCore/QJsonDocument>
+
+FunModule::FunModule() : Module("funutil", true) 
+{
+
+	RegisterCommand(Commands::FUN_MEME, "meme", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) 
+		{
+
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+
+		if (args.first() != prefix + "meme")
+			return;
+
+		if (args.size() >= 2) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+
+		auto guild = channel.id();
+		QNetworkAccessManager* manager = new QNetworkAccessManager();
```

Comment:
This is a memory leak. What you can do is store the manager in the FunModule class as a member so we don't have to create the entire subsystem each time someone invokes the !meme comand!

Reply:
Whoops... Thanks for pointing this out!